backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java

```diff
@@ -1184,18 +1192,31 @@ public GetResourcesResponseItem getEntityResource(
   }
 
   @Override
-  public List<GetResourcesResponseItem> getResourceItems(Optional<Resources> filterTo) {
+  public List<GetResourcesResponseItem> getResourceItems(
+      List<String> resourceIds, ModelDBServiceResourceTypes modelDBServiceResourceTypes) {
     try (AuthServiceChannel authServiceChannel = new AuthServiceChannel()) {
-      final GetResources.Builder getResourcesBuilder = GetResources.newBuilder();
-      if (filterTo.isPresent()) {
-        getResourcesBuilder.setResources(filterTo.get());
-      }
+      ResourceType resourceType =
+          ResourceType.newBuilder()
+              .setModeldbServiceResourceType(modelDBServiceResourceTypes)
+              .build();
+      Resources resources =
+          Resources.newBuilder()
+              .setResourceType(resourceType)
+              .setService(ServiceEnum.Service.MODELDB_SERVICE)
+              .addAllResourceIds(resourceIds)
+              .build();
 
       final GetResources.Response response =
           authServiceChannel
               .getCollaboratorServiceBlockingStub()
-              .getResources(getResourcesBuilder.build());
-      return response.getItemList();
+              .getResources(GetResources.newBuilder().setResources(resources).build());
+      return response.getItemList().stream()
+          .filter(
+              item ->
+                  item.getResourceType().getModeldbServiceResourceType()
+                          == modelDBServiceResourceTypes
+                      && item.getService() == ServiceEnum.Service.MODELDB_SERVICE)
```

Comment:
If this is not equal, then it's a bug on UAC, right? Why did you feel the need to add this check?

Reply:
Yes. But I have added it for the same side.

Reply:
I will remove this.